libchisel/src/dropsection.rs

```diff
@@ -0,0 +1,116 @@
+use super::{ModuleError, ModuleTranslator};
+
+use parity_wasm::builder::*;
+use parity_wasm::elements::*;
+
+/// Enum on which ModuleTranslator is implemented.
+pub enum DropSection<'a> {
+    NamesSection,
+    /// Name of the custom section.
+    CustomSectionByName(&'a String),
+    /// Index of the custom section.
+    CustomSectionByIndex(usize),
+    /// Index of the unknown section.
+    UnknownSectionByIndex(usize),
+}
+
+fn names_section_index_for(module: &Module) -> Result<usize, String> {
+    for (index, section) in module.sections().iter().enumerate() {
+        if let Section::Name(_section) = section {
+            return Ok(index);
+        }
+    }
+    Err("Not found".to_string())
+}
+
+fn custom_section_index_for(module: &Module, name: &String) -> Result<usize, String> {
+    for (index, section) in module.sections().iter().enumerate() {
+        if let Section::Custom(_section) = section {
+            if _section.name() == name {
+                return Ok(index);
+            }
+        }
+    }
+    Err("Not found".to_string())
+}
+
+impl<'a> DropSection<'a> {
+    fn find_index(&self, module: &Module) -> Result<usize, String> {
+        Ok(match &self {
+            DropSection::NamesSection => names_section_index_for(module)?,
+            DropSection::CustomSectionByName(name) => custom_section_index_for(module, &name)?,
+            DropSection::CustomSectionByIndex(index) => *index,
+            DropSection::UnknownSectionByIndex(index) => *index,
+        })
+    }
+
+    fn drop_section(&self, module: &mut Module) -> Result<bool, ModuleError> {
+        let index = self.find_index(&module)?;
+
+        let sections = module.sections_mut();
+        if index > sections.len() {
+            return Err(ModuleError::Custom("Not found.".to_string()));
```

Comment:
Not sure if this should be considered an error or marked as `Ok(false)`.

Potentially we need a flag in the translator configuration saying whether fields not found is an error or not and behave accordingly.

Reply:
Since there is no case where `Ok(false)` is returned and `false` indicates no mutation, it probably makes sense that this should be OK. Although a flag would make it more versatile.